src/src/PluginStructs/P104_data_struct.cpp

```diff
@@ -0,0 +1,2458 @@
+  int saveOffset = 0;
+
+  numDevices = 0;                                           // Count the number of connected display units
+
+  bufferSize = P104_CONFIG_VERSION_V2;                      // Save special marker that we're using V2 settings
+  // This write is counting
+  SaveToFile(SettingsType::Enum::CustomTaskSettings_Type, taskIndex, (uint8_t *)&bufferSize, sizeof(bufferSize), saveOffset);
+  saveOffset += sizeof(bufferSize);
+
+  if (zbuffer.reserve(P104_SETTINGS_BUFFER_V2 + 2)) {
+    for (auto it = zones.begin(); it != zones.end() && error.length() == 0; ++it) {
+      zbuffer = EMPTY_STRING;
+
+      // WARNING: Order of values should match the numeric order of P104_OFFSET_* values
+      zbuffer += it->size;          // 2
+      zbuffer += P104_FIELD_SEP;    // 1
+      zbuffer += it->text;          // 2 + ~15
+      zbuffer += P104_FIELD_SEP;    // 1
+      zbuffer += it->content;       // 1
+      zbuffer += P104_FIELD_SEP;    // 1
+      zbuffer += it->alignment;     // 1
+      zbuffer += P104_FIELD_SEP;    // 1
+      zbuffer += it->animationIn;   // 2
+      zbuffer += P104_FIELD_SEP;    // 1
+      zbuffer += it->speed;         // 5
+      zbuffer += P104_FIELD_SEP;    // 1
+      zbuffer += it->animationOut;  // 2
+      zbuffer += P104_FIELD_SEP;    // 1
+      zbuffer += it->pause;         // 5
+      zbuffer += P104_FIELD_SEP;    // 1
+      zbuffer += it->font;          // 1
+      zbuffer += P104_FIELD_SEP;    // 1
+      zbuffer += it->layout;        // 1
+      zbuffer += P104_FIELD_SEP;    // 1
+      zbuffer += it->specialEffect; // 1
+      zbuffer += P104_FIELD_SEP;    // 1
+      zbuffer += it->offset;        // 2
+      zbuffer += P104_FIELD_SEP;    // 1
+      zbuffer += it->brightness;    // 2
+      zbuffer += P104_FIELD_SEP;    // 1
+      zbuffer += it->repeatDelay;   // 4
+      zbuffer += P104_FIELD_SEP;    // 1
+
+      // 45 total + (max) 100 characters for it->text requires a buffer of ~150 (P104_SETTINGS_BUFFER_V2), but only required length is
+      // stored with the length prefixed
+
+      numDevices += (it->size != 0 ? it->size : 1) + it->offset;                                // Count corrected for newly added zones
+
+      if (saveOffset + zbuffer.length() + (sizeof(bufferSize) * 2) > (DAT_TASKS_CUSTOM_SIZE)) { // Detect ourselves if we've reached the
+        error.reserve(55);                                                                      // high-water mark
+        error += F("Total combination of Zones & text too long to store.\n");
+        addLog(LOG_LEVEL_ERROR, error);
+      } else {
+        // Store length of buffer
+        bufferSize = zbuffer.length();
+
+        // As we write in parts, only count as single write.
+        if (RTC.flashDayCounter > 0) {
+          RTC.flashDayCounter--;
+        }
+        SaveToFile(SettingsType::Enum::CustomTaskSettings_Type, taskIndex, (uint8_t *)&bufferSize, sizeof(bufferSize), saveOffset);
```

Comment:
`bufferSize` is the `zbuffer.length()` so you can just refer to the string buffer to save it.
`static_cast<const uint8_t*>(zbuffer.c_str())`

Reply:
The compiler doesn't like that

Reply:
with plain cast you mean `(const uint_t*)zbuffer.c_str()` ?
that old C-style cast is more like a `reinterpret_cast`.